File: Challenge 7/c7_easy_bonus.py
#Daily Programmer Challenge 7 : Easy (BONUS)
#@author:   Baron Daugherty
#@date:     2015-06-15
#@desc:     A bonus extension to challenge 7 easy that translates a string of
#           english text into morse code

import winsound

#main method drives the program
def main():
    #gather the text to translate
    text = input("Please enter the text to translate: ").lower()

    #output the translation
    morse = translate(text)
    print(morse)

#translate method translates the english text to morse code, letters have a
#space between them and words are delimited by space-slash-space in accordance
#to the original challenge spec
def translate(text):
    #empty list for translated characters
    translated = []

    #dictionary is keyed on the english letter with morse code dot-dash values
    code = {'a':'.-', 'b':'-...', 'c':'-.-.', 'd':'-..', 'e':'.', 'f':'..-.',
            'g':'--.', 'h':'....', 'i':'..', 'j':'.---', 'k':'-.-', 'l':'.-..',
            'm':'--', 'n':'-.', 'o':'---', 'p':'.--.', 'q':'--.-', 'r':'.-.',
            's':'...', 't':'-', 'u':'..-', 'v':'...-', 'w':'.--', 'x':'-..-',
            'y':'-.--', 'z':'--..'}

    #go through each word and each letter and translate into morse code
    #add the letter translations to the list and add spaces and slashes
    #where appropriate
    for word in text.split(' '):
        for letter in word:
            translated.append(code[letter])
            translated.append(' ')
        translated.append('/ ')

    #return translated as a single string
    #(truncates the unnecessary ' / ' at the end)
    return ''.join(translated[:-2])

#start the program
main()

File: Challenge 7/c7_easy_bonus.pl
#!/usr/bin/perl
#   Daily Programmer Challenge 7 : Easy (Bonus)
#   @author:    Baron Daugherty
#   @date:      2015-06-28
#   @desc:      Write a program that can translate english
#               phrases into morse code
use warnings;
use strict;

#hash as an english to morse table
my %alpha = ('a'=>'.-', 'b'=>'-...', 'c'=>'-.-.', 'd'=>'-..', 'e'=>'.', 'f'=>'..-.',
            'g'=>'--.', 'h'=>'....', 'i'=>'..', 'j'=>'.---', 'k'=>'-.-', 'l'=>'.-..',
            'm'=>'--', 'n'=>'-.', 'o'=>'---', 'p'=>'.--.', 'q'=>'--.-', 'r'=>'.-.',
            's'=>'...', 't'=>'-', 'u'=>'..-', 'v'=>'...-', 'w'=>'.--', 'x'=>'-..-',
            'y'=>'-.--', 'z'=>'--..');

#list for translated characters
my @translated;

#gather the text to encode
print "Please enter the text to encode: ";
chomp(my $text = lc(<>));

#get each word, then each letter in the word and add the hash value to the list
foreach my $word (split ' ', $text){
    foreach my $letter (split //, $word){
        push(@translated, $alpha{$letter});
        push(@translated, ' ');
    }
    push(@translated, '/ ');
}

#print the list as one string
print("\n", join('', @translated));